streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -641,6 +678,19 @@ public Serde valueSerde() {
         return parsed;
     }
 
+    @Override
+    public Long getLong(final String key) {
+        // return different default value for `commit.interval.ms` if EOS enabled
+        if (StreamsConfig.COMMIT_INTERVAL_MS_CONFIG.equals(key)
+            && eosEnabled
+            // use EOS default value only if no user config is set
+            && !originals().containsKey(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG)) {
+            return EOS_DEFAULT_COMMIT_INTERVAL_MS;
```

Comment:
This approach is a potential source of confusion because if you iterate over the entries of the config, the override won't be there. I think you either want to do the override in the place where this value is used (like it's done in the producer) or it could be done by mutating the config itself. It may also be a good idea to log the override (like the producer does).